specfile/specfile.py

```diff
@@ -120,18 +183,32 @@ def _dump_debug_info(self, message) -> None:
         )
 
     @staticmethod
-    def _read_lines(path: Path) -> Tuple[List[str], bool]:
-        content = path.read_text(encoding="utf8", errors="surrogateescape")
-        return content.splitlines(), content[-1] == "\n"
+    def _read_lines(stream: TextIO) -> Tuple[List[str], bool]:
+        stream.seek(0)
+        content = stream.read()
+        return content.splitlines(), content.endswith("\n")
+
+    @property
+    def is_writable(self) -> bool:
+        return hasattr(self._file, "write") and callable(
+            getattr(self._file, "write", None)
+        )
```

Comment:
Still not sure how useful this will be, but:
- let's keep the naming consistent, there is already e.g. `tainted` property, not `is_tainted`
- `IOBase` has `writable()` method: https://docs.python.org/3/library/io.html#io.IOBase.writable

```suggestion
    def writable(self) -> bool:
        return self._file.writable()
```

Reply:
Right, seems like I got extra paranoid with the checks

Reply:
I will just remove it for now